src/cdk/tree/tree.ts

```diff
@@ -371,4 +377,22 @@ export class CdkTreeNode<T> implements FocusableOption, OnDestroy {
     }
     this.role = 'treeitem';
   }
+
+  private _getParentNodeAriaLevel(): number {
+    let parent = this._elementRef.nativeElement.parentElement;
+    while (parent &&
+    !(parent.classList.contains('cdk-nested-tree-node') || parent.classList.contains('cdk-tree'))) {
+      parent = parent.parentElement;
+    }
+    if (!parent) {
+      throw Error('Incorrect tree structure containing detached node.');
+    }
+    if (parent.classList.contains('cdk-nested-tree-node')) {
+      return parseInt(parent.getAttribute('aria-level')!);
+    } else if (parent.classList.contains('cdk-tree')) {
+      return 0;
+    } else {
+      throw Error(`Incorrect tree structure containing ${parent.className}.`);
```

Comment:
I'm not sure this error message makes sense

Reply:
Since the only cases outside the while loop are `!parent`, `parent.classList.contains('cdk-nested-tree-node')` and `parent.classList.contains('cdk-nested-tree-node')` I decided to put them together in and if - else if - else. Having the explicit cases is more clear but that else clause should be unreachable and having an error message like "unreachable code" seems like bad practice. wdyt?